api/src/controllers/put/mine.js

```diff
@@ -0,0 +1,57 @@
+const mongoose = require('mongoose');
+const ObjectId = require('mongoose').Types.ObjectId;
+
+const SYSTEM_USER = 'SYSTEM_USER';
+
+/**
+ * Performs all operations necessary to edit a master Mine record.
+ *
+ * @param {*} args
+ * @param {*} res
+ * @param {*} next
+ * @param {*} incomingObj see example
+ * @returns object containing the operation's status and created records
+ */
+exports.editRecord = async function(args, res, next, incomingObj) {
+  try {
+    if (incomingObj._schemaName !== 'MineBCMI') {
+      throw new Error('editRecord - incorrect schema type, must be MineBCMI');
+    }
+
+    const Model = mongoose.model('MineBCMI');
+
+    const record = await Model.findOneAndUpdate(
+        { _id: new ObjectId(incomingObj._id) },
+        { $set: {
+            name: incomingObj.name,
+            permitNumbers: incomingObj.permitNumbers,
+            status: incomingObj.status,
+            commodities: incomingObj.commodities,
```

Comment:
Rather than manually setting every value, there is some santize/dotnotation code (see other put controllers) that make it a bit simpler to safely set everything, including nested objects.

Reply:
Updated to use existing helper functions.